Gems/GenAIFramework/Code/Include/GenAIFramework/Feature/JsonUtils.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <GenAIFramework/Communication/AIModelRequestBus.h>
+
+#include <AzCore/JSON/document.h>
+#include <AzCore/JSON/rapidjson.h>
+#include <AzCore/JSON/writer.h>
+
+namespace GenAIFramework::JsonUtils
+{
+    inline rapidjson::Value AIMessageToJsonValue(const AIMessage& message, rapidjson::Document::AllocatorType& allocator)
+    {
+        rapidjson::Value AIMessageJson(rapidjson::kObjectType);
+
+        rapidjson::Value entryJson(rapidjson::kObjectType);
+
+        AZStd::string role;
+
+        switch (message.first)
+        {
+        case Role::User:
+            role = "user";
+            break;
+        case Role::Assistant:
+            role = "assistant";
+            break;
+        case Role::System:
+            role = "system";
+            break;
+        }
+
+        rapidjson::Value roleValue;
+        roleValue.SetString(role.c_str(), allocator);
+        AIMessageJson.AddMember("role", roleValue, allocator);
+
+        rapidjson::Value contentJson(rapidjson::kArrayType);
+        for (const auto& content : message.second)
+        {
+            rapidjson::Value contentValue;
+            if (content.type() == AZ::AzTypeInfo<AZStd::string>().Uuid())
+            {
+                contentValue.SetString(AZStd::any_cast<AZStd::string>(content).c_str(), allocator);
+            }
+
+            contentJson.PushBack(contentValue, allocator);
```

Comment:
Does it make sense to push empty values?
```suggestion
            if (content.type() == AZ::AzTypeInfo<AZStd::string>().Uuid())
            {
                rapidjson::Value contentValue;
                contentValue.SetString(AZStd::any_cast<AZStd::string>(content).c_str(), allocator);
                contentJson.PushBack(contentValue, allocator);
            }
```